pages/star-search/waitlist.tsx

```diff
@@ -0,0 +1,176 @@
+            style={{ height: "40px" }}
+          >
+            <Image
+              src="/assets/star-search-logo-white.svg"
+              alt="StarSearch logo"
+              width={24}
+              height={24}
+              className="w-6 h-6"
+            />
+          </div>
+        );
+    }
+  };
+
+  const extraClasses = author === "You" ? "p-[14px] rounded-[12px]" : "py-[34px] px-[30px] rounded-[22px]";
+
+  return (
+    <>
+      {author === "You" ? renderAvatar() : null}
+      <div className={`waitlist-chatbox bg-white w-max border-[2px] border-gradient-orange-one ${extraClasses}`}>
+        <Markdown className="grid gap-4 [&_ol]:grid [&_ol]:gap-4 max-w-lg">{content}</Markdown>
+      </div>
+      {author === "StarSearch" ? renderAvatar() : null}
+    </>
+  );
+}
+
+export const getServerSideProps = async (context: GetServerSidePropsContext) => {
+  const supabase = createPagesServerClient(context);
+  const {
+    data: { session },
+  } = await supabase.auth.getSession();
+
+  let isWaitlisted = false;
+  let waitlistCount = 0;
+
+  if (session) {
+    const [waitlistCountResponse, userResponse] = await Promise.all([
+      fetch(`${process.env.NEXT_PUBLIC_API_URL}/auth/waitlisted`),
+      fetch(`${process.env.NEXT_PUBLIC_API_URL}/users/${session.user.user_metadata.user_name}`, {
+        headers: {
+          accept: "application/json",
+        },
+      }),
+    ]);
+
+    const [waitlistPayload, user] = await Promise.all([waitlistCountResponse.json(), userResponse.json()]);
+
+    isWaitlisted = user.is_waitlisted;
+    waitlistCount = waitlistPayload.waitlisted_users;
+  }
+
+  return {
+    props: {
+      isWaitlisted,
+      waitlistCount,
+    },
+  };
+};
+
+interface StarSearchWaitListPageProps {
+  isWaitlisted: boolean;
+  waitlistCount: number;
+}
+
+export default function StarSearchWaitListPage({ isWaitlisted, waitlistCount }: StarSearchWaitListPageProps) {
+  const { sessionToken, signIn, user } = useSupabaseAuth();
+
+  async function joinWaitlist(event: React.FormEvent<HTMLFormElement>) {
+    event.preventDefault();
+    const bearerToken = sessionToken ? sessionToken : undefined;
+    const response = await fetch(`${process.env.NEXT_PUBLIC_API_URL}/auth/waitlist`, {
+      method: "POST",
+      headers: {
+        "Accept-Type": "application/json",
+        Authorization: `Bearer ${bearerToken}`,
+      },
+    });
+
+    return response.status === 201;
+  }
+
+  return (
+    <ProfileLayout>
+      <div className="flex flex-col items-center gap-4 mb-16">
+        <div className="flex gap-2 items-center">
+          <Image src="/assets/star-search-logo.svg" alt="" width={40} height={40} />
+          <h1 className="text-3xl lg:text-4xl font-bold bg-clip-text text-transparent bg-gradient-to-r from-sauced-orange to-amber-400">
+            StarSearch
+          </h1>
+        </div>
+        <p className="font-semibold text-5xl">Copilot, but for your git history</p>
```

Comment:
```suggestion
        <p className="font-semibold text-5xl">Copilot, but for git history</p>
```

Seeing on screen, I think maybe we remove your to broaden the ideal use case. Thoughts @isabensusan?